be/src/io/s3_input_stream.cpp

```diff
@@ -22,6 +24,11 @@ StatusOr<int64_t> S3InputStream::read(void* out, int64_t count) {
     request.SetBucket(_bucket);
     request.SetKey(_object);
     request.SetRange(std::move(range));
+    if (config::object_storage_request_payer_enabled) {
+        request.SetRequestPayer(Aws::S3::Model::RequestPayer::requester);
```

Comment:
Actually I have some question about it.
When we setting this config without setting other configs, how does S3 know who I am?
second, I think this configure is better to set more high level. If we set the the config here, all of the request will be the charged by the requester. I'm afraid that it is not suitable for our future usage.

Reply:
Depends on which ec2(or container) you are running at I guess. 

I also have this concern, but I don't know where to add different options. Looks following code is the right place, however `RandomAccessFileOptions` is shared by all filessystems.

```
struct RandomAccessFileOptions {
    RandomAccessFileOptions() = default;
};


StatusOr<std::unique_ptr<RandomAccessFile>> S3FileSystem::new_random_access_file(const RandomAccessFileOptions& opts,
                                                                                 const std::string& path) {
    S3URI uri;
    if (!uri.parse(path)) {
        return Status::InvalidArgument(fmt::format("Invalid S3 URI: {}", path));
    }
    auto client = new_s3client(uri);
    auto input_stream = std::make_shared<io::S3InputStream>(std::move(client), uri.bucket(), uri.key());
    return std::make_unique<RandomAccessFile>(std::move(input_stream), path);
}
```

Reply:
Maybe this field should be put into FE session variable, and passed down to BE in plan ?